### ISSUE TITLE
Bad Optimization

### ISSUE THREAD

**Issue by Nelsonvon** [dalab/end2end_neural_el#8]
hello,
I utilize this software on my won corpus, which are tagged and converted to the form the same as AIDA datasets.
However, I found that this software has a high opportunity to fall into bad local optimum. With the same training setting suggested by the author, about 50% of all my training task eventually terminated with f1 score less than 20. When I increase the learning rate five times( lr = 0.005 ), it would be better but the bad optimum still exists sometimes. 
Do anyone meet the same problem and are willing to share how to handle it?
thx

**Comment by NikosKolitsas** [comment]
Is there a case that your corpus is smaller than the AIDA? A few things that come to my mind is to reduce the number of variables of the neural network (size of hidden states of the LSTMs etc), and also to maybe change the termination criterion (I stop the training when after three checks the dev score hasn't improved significantly, maybe increase this number of how often you evaluate to dev set).
